shared/src/main/scala/mlscript/ConstraintSolver.scala

```diff
@@ -602,6 +617,13 @@ class ConstraintSolver extends NormalForms { self: Typer =>
           tv.lowerBounds ::= nv
           nv.upperBounds = tv.upperBounds.map(extrude(_, lvl, pol))
         }
+        nv.indexedBy = tv.indexedBy
+        nv.indexedIn = tv.indexedIn
+
+        nv.indexedBy.foreach { case (index, result) =>
+          { extrude(index, lvl, pol); extrude(result, lvl, pol) } }
+        nv.indexedIn.foreach { case (receiver, result) =>
+          {extrude(receiver, lvl, pol); extrude(result, lvl, pol)} }
```

Comment:
This logic is incorrect. `extrude` does not do anything to its argument, it's the _result_ of it that should be used.

It should look like this:

```scala
nv.indexedBy = nv.indexedBy.map { case (index, result) =>
  (extrude(index, lvl, pol), extrude(result, lvl, pol)) }
```